distribution/lib/Standard/Base/0.0.0-dev/src/Data/Statistics.enso

```diff
@@ -388,3 +344,78 @@ is_valid v = case v of
     Nothing -> False
     _ : Number -> v.is_nan.not
     _ -> True
+
+## PRIVATE
+type Accumulator
+    new : Integer -> Accumulator
+    new moments=0 = Accumulator.Value 0 Nothing Nothing (Vector.fill moments 0)
+
+    ## PRIVATE
+    Value count:Integer minimum:Any maximum:Any (moments:Vector|Nothing)
+
+    ## PRIVATE
+    comparator_error : Boolean
+    comparator_error self = self.minimum.is_nothing
+
+    ## PRIVATE
+    increment self value = if self.count == 0 then (Accumulator.Value 1 value value (self.update_moments value)) else
+        minimum = Accumulator.comparator self.minimum value Ordering.Less
+        maximum = Accumulator.comparator self.maximum value Ordering.Greater
+        Accumulator.Value self.count+1 minimum maximum (self.update_moments value)
+
+    ## PRIVATE
+    comparator current value bound = if current == Nothing then Nothing else
+        comp = Ordering.compare value current . catch Incomparable_Values (_-> Nothing)
+        if comp == Nothing then Nothing else
+            if comp == bound then value else current
```

Comment:
While definitely not a good thing to happen, I think it is possible for a custom comparator to throw other errors than just `Incomparable_Values`.

For example
```
from Standard.Base import all
import Standard.Base.Errors.Illegal_State.Illegal_State

type My_Type
    Value x

type My_Comparator
    compare x y =
        _ = [x, y]
        Error.throw (Illegal_State.Error "TEST")
    hash x = x

Comparable.from (_:My_Type) = My_Comparator

main =
    x = My_Type.Value 1
    y = My_Type.Value 2
    IO.println (x < y)
```
will result in:
```
(Error: (Illegal_State.Error 'TEST' Nothing))
```

This has the consequence that such an error like shown here will propagate to the fold result. That fold result is ignored in `running_bulk` (so the error will be swallowed) but it will be propagated in `compute_bulk` (since we inspect `counter.count` in an `if` and if that's an error it will propagate).

1. I think we should avoid propagating comparison errors if Min/Max was not requested.
2. I think we should propagate comparison errors consistently between `running_bulk` and `compute_bulk`.

The (1) may not actually be a problem because I assume whenever we compute other statistics (relying on moments), we are already in the 'numeric' world where all comparisons should be builtin and never error.

But (2) is a real issue when computing a Min/Max of custom objects.

Reply:
This should now be handled correctly. The computation will fail if asked for Min/Max and it gets any error computing. The result will be the error.